pkg/appgw/frontend_listeners.go

```diff
@@ -74,9 +74,9 @@ func (builder *appGwConfigBuilder) getPublicIPID() *string {
 	for _, ip := range *builder.appGwConfig.FrontendIPConfigurations {
 		// Collect the JSON IP configs for debug purposes.
 		if jsonConf, err := ip.MarshalJSON(); err != nil {
-			jsonConfigs = append(jsonConfigs, string(jsonConf))
-		} else {
 			glog.Error("Could not marshall IP configuration:", *ip.ID, err)
```

Comment:
Minor, Will it ever happen that we are unable to marshal ?
If yes, how are we able to use the ip object below if it was not a valid de-serialized object in the first case?

Reply:
First, I think that we would be unable to marshal only if the SDK allowed for some bug to sneak in.  I don't expect to ever run into that scenario, but since it is theoretically possible - we handle it.

And if that ever happens - we would simply not append the unmarshallable IP into the list - so IF we actually end up at line https://github.com/Azure/application-gateway-kubernetes-ingress/pull/197/files#diff-b08d8c4173cd5b924fa92404420f18deR94 - we may not have listed an IP address that we failed to marshall.

Overall if we ever find ourselves checking off both of these boxes (couldn't unmarshall IP and could not  find a public IP) - we would have sufficient log coverage to help us troubleshoot and understand why this happened.